src/auth0-session/transient-store.ts

```diff
@@ -10,70 +10,49 @@ export interface StoreOptions {
   value?: string;
 }
 
-const header = { alg: 'HS256', b64: false, crit: ['b64'] };
-const getPayload = (cookie: string, value: string): Buffer => Buffer.from(`${cookie}=${value}`);
-const flattenedJWSFromCookie = (cookie: string, value: string, signature: string): JWS.FlattenedJWS => ({
-  protected: Buffer.from(JSON.stringify(header))
-    .toString('base64')
-    .replace(/=/g, '')
-    .replace(/\+/g, '-')
-    .replace(/\//g, '_'),
-  payload: getPayload(cookie, value),
-  signature
-});
-const generateSignature = (cookie: string, value: string, key: JWK.Key): string => {
-  const payload = getPayload(cookie, value);
-  return JWS.sign.flattened(payload, key, header).signature;
-};
-const verifySignature = (cookie: string, value: string, signature: string, keystore: JWKS.KeyStore): boolean => {
-  try {
-    return !!JWS.verify(flattenedJWSFromCookie(cookie, value, signature), keystore, {
-      algorithms: ['HS256'],
-      crit: ['b64']
-    });
-  } catch (err) {
-    return false;
-  }
-};
-const getCookieValue = (cookie: string, value: string, keystore: JWKS.KeyStore): string | undefined => {
-  if (!value) {
+const getCookieValue = async (k: string, v: string, keys: Uint8Array[]): Promise<string | undefined> => {
+  if (!v) {
     return undefined;
   }
-  let signature;
-  [value, signature] = value.split('.');
-  if (verifySignature(cookie, value, signature, keystore)) {
-    return value;
+  const [value, signature] = v.split('.');
+  const flattenedJWS = {
+    protected: jose.base64url.encode(JSON.stringify({ alg: 'HS256', b64: false, crit: ['b64'] })),
+    payload: `${k}=${value}`,
+    signature
+  };
+  for (let key of keys) {
+    try {
+      await jose.flattenedVerify(flattenedJWS, key, {
+        algorithms: ['HS256'],
+        crit: {
+          b64: false
+        }
```

Comment:
I believe passing this option to the verify API is not necessary.

https://github.com/panva/jose/blob/v4.9.2/docs/interfaces/types.VerifyOptions.md

> The JWS extension Header Parameter `b64` is always recognized and processed properly. No other registered Header Parameters that need this kind of default built-in treatment are currently available.

Reply:
👍 5aa49c3